.github/workflows/synchronize_submodules.yml

```diff
@@ -12,7 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Sets submodule state to the one specified in SUBMODULE_VERSIONS
+# Edits the HEAD commit to use the submodule state specified in SUBMODULE_VERSIONS
+# WARNING: rewrites history!
```

Comment:
Might be worth an extra warning on line 23 where branches are listed so someone isn't tempted to enable it on a branch humans use like `main`.

Reply:
Done.